shell/platform/darwin/ios/framework/Source/SemanticsObject.mm

```diff
@@ -294,23 +294,22 @@ - (instancetype)initWithBridge:(fml::WeakPtr<flutter::AccessibilityBridgeIos>)br
     _bridge = bridge;
     _uid = uid;
     _children = [[NSMutableArray alloc] init];
-    _childrenInHitTestOrder = [[NSMutableArray alloc] init];
+    _childrenInHitTestOrder = [[NSArray alloc] init];
   }
 
   return self;
 }
 
 - (void)dealloc {
   for (SemanticsObject* child in _children) {
-    [child privateSetParent:nil];
+    child.parent = nil;
   }
   [_children removeAllObjects];
-  [_childrenInHitTestOrder removeAllObjects];
   [_children release];
   [_childrenInHitTestOrder release];
 
   _parent = nil;
```

Comment:
Is this here for the same reason as the awful `_inDealloc` workaround? In theory it shouldn't be necessary.

Reply:
The `_inDealloc` workaround was done in https://github.com/flutter/engine/pull/27786, which didn't include setting `_parent` to nil.  

Setting `_parent` to nil was done in #4602 and I think it was for this guard:
https://github.com/flutter/engine/blob/b4b798d2e706cf360e3a075bbe818e85a4634c7b/shell/platform/darwin/ios/framework/Source/SemanticsObject.mm#L709-L714

I don't like this but it seems like iOS is, or at least was, digging around the `accessibilityContainer` type methods during `-[super dealloc]` which was causing crashes.  I'd rather not touch that for this refactor.